core/src/main/java/tachyon/conf/TachyonConf.java

```diff
@@ -0,0 +1,378 @@
+    this(true);
+  }
+
+  /**
+   * Test constructor for TachyonConfTest class.
+   *
+   * Most clients will call this constructor to allow default loading of properties to happen.
+   *
+   */
+  TachyonConf(boolean includeSystemProperties) {
+    loadDefault(includeSystemProperties);
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = 0;
+    for (Object s : mProperties.keySet()) {
+      hash ^= s.hashCode();
+    }
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    } else if (obj instanceof TachyonConf) {
+      Properties props = ((TachyonConf) obj).getInternalProperties();
+      return mProperties.equals(props);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Here is the order of importance of resource where we load the properties:
+   *   -) System properties if desired
+   *   -) Site specific properties via tachyon-site.properties file
+   *   -) Default properties via tachyon-default.properties file
+   * so we will load it in reverse order.
+  */
+  protected void loadDefault(boolean includeSystemProperties) {
+    // Load default
+    Properties defaultProps = new Properties();
+
+    // Override runtime default
+    defaultProps.setProperty(Constants.MASTER_HOSTNAME, NetworkUtils.getLocalHostName());
+    defaultProps.setProperty(Constants.WORKER_NETWORK_NETTY_CHANNEL,
+        ChannelType.defaultType().toString());
+    defaultProps.setProperty(Constants.WORKER_SERVER_THREADS,
+        String.valueOf(Runtime.getRuntime().availableProcessors()));
+    defaultProps.setProperty(Constants.MASTER_SERVER_THREADS,
+        String.valueOf(2 * Runtime.getRuntime().availableProcessors()));
+
+    InputStream defaultInputStream =
+        TachyonConf.class.getClassLoader().getResourceAsStream(DEFAULT_PROPERTIES);
+    if (defaultInputStream != null) {
+      try {
+        defaultProps.load(defaultInputStream);
+      } catch (IOException e) {
+        throw new RuntimeException("Unable to load default Tachyon properties file.");
+      }
+    }
+
+    // Update tachyon.master_address
+    String masterHostname = defaultProps.getProperty(Constants.MASTER_HOSTNAME);
+    String masterPort = defaultProps.getProperty(Constants.MASTER_PORT);
+    boolean useZk = Boolean.parseBoolean(defaultProps.getProperty(Constants.USE_ZOOKEEPER));
+    String masterAddress = (useZk ? Constants.HEADER_FT : Constants.HEADER) + masterHostname + ":"
+        + masterPort;
+    defaultProps.setProperty(Constants.MASTER_ADDRESS, masterAddress);
+
+    // Load site specific properties file
+    Properties siteProps = new Properties();
+    InputStream siteInputStream =
+        TachyonConf.class.getClassLoader().getResourceAsStream(SITE_PROPERTIES);
+    if (siteInputStream != null) {
+      try {
+        siteProps.load(siteInputStream);
+      } catch (IOException e) {
+        LOG.info("Unable to load site Tachyon configuration file.", e);
```

Comment:
It might be slightly confusing if an info log dumps a stack trace, but I'm not sure if there's a better way (maybe warn level? since the file is present just failed to be read).